src/NUnitFramework/framework/Internal/TestFilter.cs

```diff
@@ -156,15 +156,15 @@ public static TestFilter FromXml(TNode node)
             {
                 case "filter":
                 case "and":
-                    List<TestFilter> childFilters = new List<TestFilter>();
+                    var childFilters = new List<TestFilter>(node.ChildNodes.Count);
```

Comment:
Nice touch to pre-initialize this. Thanks!

nitpick: Since we have to convert it to an array a few lines below, would it make sense to make this an array here instead of a list to save an array copy operation? Similar suggest for the "or" below.

Reply:
I agree with your nitpick, I should have looked how the list was used after.
Changed code to create array, extracted to a function so I only needed one copy.